x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/edit_output_flyout/advanced_options_section.tsx

```diff
@@ -0,0 +1,303 @@
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+        <EuiFormRow
+          fullWidth
+          {...queueFlushTimeout.formRowProps}
+          label={
+            <FormattedMessage
+              id="xpack.fleet.settings.editOutputFlyout.queueFlushTimeoutLabel"
+              defaultMessage="Flush Timeout"
+            />
+          }
+        >
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiFieldNumber {...queueFlushTimeout.props} placeholder="Seconds" min={0} />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.queueFlushTimeoutDescription"
+                  defaultMessage="Upon expiry the output queue is flushed and data is written to the output."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+        <EuiFormRow
+          fullWidth
+          {...memQueueEvents.formRowProps}
+          label={
+            <FormattedMessage
+              id="xpack.fleet.settings.editOutputFlyout.memQueueEventsLabel"
+              defaultMessage="Memory Queue Events"
+            />
+          }
+        >
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiFieldNumber {...memQueueEvents.props} placeholder="Events" min={0} />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.memQueueEventsSizeDescription"
+                  defaultMessage="Maximum number of events that can be stored in the queue. Default is set to 4096."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+
+        <EuiHorizontalRule />
+
+        <EuiFormRow fullWidth {...diskQueueEnabledInput.formRowProps}>
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiSwitch
+                data-test-subj="editOutputFlyout.diskQueueSwitch"
+                {...diskQueueEnabledInput.props}
+                label={
+                  <FormattedMessage
+                    id="xpack.fleet.settings.editOutputFlyout.diskQueueSwitchLabel"
+                    defaultMessage="Disk Queue"
+                  />
+                }
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.diskQueueSwitchDescription"
+                  defaultMessage="Once enabled, events will be queued on disk when there's a requirement to cache them on the agents in the event of connection loss."
```

Comment:
Should we be this specific about why a user would want to use this? I don't think it's only useful for connection loss but also when there may be a large burst of data that can't fit in memory.

We should also indicate how this is used in conjunction with the memory queue. One question I think we'll get: Does enabling the disk queue turn off the memory queue?

Reply:
I don't believe Memory Queue is disabled (@cmacknz and @leehinman could you conform please)

@joshdover @criamico yes I agree with the comment, let's make the copy more generic: "Once enabled, events will be queued on disk if, for some reason, agent is not able to send them."